oxd-server/src/main/java/org/gluu/oxd/server/mapper/RegisterResponseMapper.java

```diff
@@ -0,0 +1,447 @@
+            rpFromRegisterResponse.setRequestObjectSigningAlg(respClaims.get(REQUEST_OBJECT_SIGNING_ALG.toString()));
+        }
+
+        if (!Strings.isNullOrEmpty(respClaims.get(REQUEST_OBJECT_ENCRYPTION_ALG.toString())) &&
+                KeyEncryptionAlgorithm.fromName(respClaims.get(REQUEST_OBJECT_ENCRYPTION_ALG.toString())) != null) {
+            rpFromRegisterResponse.setRequestObjectEncryptionAlg(respClaims.get(REQUEST_OBJECT_ENCRYPTION_ALG.toString()));
+        }
+
+        if (!Strings.isNullOrEmpty(respClaims.get(REQUEST_OBJECT_ENCRYPTION_ENC.toString())) &&
+                BlockEncryptionAlgorithm.fromName(respClaims.get(REQUEST_OBJECT_ENCRYPTION_ENC.toString())) != null) {
+            rpFromRegisterResponse.setRequestObjectEncryptionEnc(respClaims.get(REQUEST_OBJECT_ENCRYPTION_ENC.toString()));
+        }
+
+        rpFromRegisterResponse.setDefaultMaxAge(safeToNumber(respClaims.get(DEFAULT_MAX_AGE.toString())));
+
+        if (!Strings.isNullOrEmpty(respClaims.get(REQUIRE_AUTH_TIME.toString()))) {
+            rpFromRegisterResponse.setRequireAuthTime(Boolean.valueOf(respClaims.get(REQUIRE_AUTH_TIME.toString())));
+        }
+
+        rpFromRegisterResponse.setInitiateLoginUri(respClaims.get(INITIATE_LOGIN_URI.toString()));
+        rpFromRegisterResponse.setAuthorizedOrigins(convertStringToList(respClaims.get(AUTHORIZED_ORIGINS.toString())));
+        rpFromRegisterResponse.setAccessTokenLifetime(safeToNumber(respClaims.get(ACCESS_TOKEN_LIFETIME.toString())));
+        rpFromRegisterResponse.setSoftwareId(respClaims.get(SOFTWARE_ID.toString()));
+        rpFromRegisterResponse.setSoftwareVersion(respClaims.get(SOFTWARE_VERSION.toString()));
+        rpFromRegisterResponse.setSoftwareStatement(respClaims.get(SOFTWARE_STATEMENT.toString()));
+        rpFromRegisterResponse.setClientJwksUri(respClaims.get(JWKS_URI.toString()));
+        rpFromRegisterResponse.setClaimsRedirectUri(convertStringToList(respClaims.get(CLAIMS_REDIRECT_URIS.toString())));
+
+        return rpFromRegisterResponse;
+    }
+
+    private static List<String> convertStringToList(String input) {
+        if (input == null)
+            return null;
+
+        String commaSeparatedString = input.replaceAll("\\[", "").replaceAll("\\]", "").replace("\"", "");
+        return Lists.newArrayList(commaSeparatedString.split(","));
+    }
+
+    private static List<String> convertSpaceSeparatedStringToList(String input) {
+        if (input == null)
+            return null;
+
+        return Lists.newArrayList(input.split(" "));
+    }
+
+    public static Integer safeToNumber(String num) {
+        if (num == null)
+            return null;
+        try {
+            return Integer.valueOf(num);
+        } catch (NumberFormatException ex) {
+            return null;
+        }
+    }
+
+    public static boolean compareParams(List<String> oxdRp, List<String> oxAuthRp) {
```

Comment:
If method is named `compare` then it usually returns `int` because author of such method should let know whether object is less/greater/equals (-1/0/1). See example from jdk `java.lang.Integer#compare`. If we return true/false then it's definitely `equals`. `isListsEqualIgnoringOrder`

Reply:
Done